cap-primitives/src/yanix/common/fs/symlink_unchecked.rs

```diff
@@ -0,0 +1,24 @@
+use std::{
+    ffi::OsStr,
+    fs, io,
+    path::Path,
+    os::unix::io::AsRawFd,
+};
+use yanix::file::symlinkat;
+
+/// *Unsandboxed* function similar to `symlink`, but which does not perform sandboxing.
+pub(crate) fn symlink_unchecked(
+    old_path: &Path,
+    new_start: &fs::File,
+    new_path: &Path,
+) -> io::Result<()> {
+    // POSIX's `symlinkat` with an empty path returns `ENOENT`, so use "." instead.
+    let new_path = if new_path.components().next().is_none() {
```

Comment:
FWIW I think `new_path.as_os_str().is_empty()` communicates the intent better here.